### PR TITLE
yoctifyfurther

### PR DESCRIPTION
**Pull request by fregante**
still2big

### CODE REVIEW

**Comment by fregante** [comment]
- uses 39 as the default second parameter
- replace const with let to save 2 characters

**Comment by smashah** [comment]
You should just remove all but two colors so it satisfies the requirement of multiple colors

File: index.js [hunk 1]
```diff
@@ -1,37 +1 @@
...
-export const bgGreen = format(42, 49);
-export const bgYellow = format(43, 49);
-export const bgBlue = format(44, 49);
-export const bgMagenta = format(45, 49);
-export const bgCyan = format(46, 49);
-export const bgWhite = format(47, 49);
-export const bgGray = format(100, 49);
+import tty from 'node:tty';let hasColors=tty.WriteStream.prototype.hasColors();let f=(startCode,endCode=39)=>hasColors?string=>'\u001B['+startCode+'m'+string+'\u001B['+endCode+'m':string=>string;export let reset=f(0,0);export let bold=f(1,22);export let dim=f(2,22);export let italic=f(3,23);export let underline=f(4,24);export let overline=f(53,55);export let inverse=f(7,27);export let hidden=f(8,28);export let strikethrough=f(9,29);export let black=f(30);export let red=f(31);export let green=f(32);export let yellow=f(33);export let blue=f(34);export let magenta=f(35);export let cyan=f(36);export let white=f(37);export let gray=f(90);export let bgBlack=f(40,49);export let bgRed=f(41,49);export let bgGreen=f(42,49);export let bgYellow=f(43,49);export let bgBlue=f(44,49);export let bgMagenta=f(45,49);export let bgCyan=f(46,49);export let bgWhite=f(47,49);export let bgGray=f(100,49)
```
Comment (FlorianWendelborn) [inline index.js:1]: #### -72B:

```suggestion
import t from 'node:tty';let h=t.WriteStream.prototype.hasColors(),f=(s,e=39)=>h?v=>'\u001B['+s+'m'+v+'\u001B['+e+'m':v=>v;export let reset=f(0,0);export let bold=f(1,22);export let dim=f(2,22);export let italic=f(3,23);export let underline=f(4,24);export let overline=f(53,55);export let inverse=f(7,27);export let hidden=f(8,28);export let strikethrough=f(9,29);export let black=f(30);export let red=f(31);export let green=f(32);export let yellow=f(33);export let blue=f(34);export let magenta=f(35);export let cyan=f(36);export let white=f(37);export let gray=f(90);export let bgBlack=f(40,49);export let bgRed=f(41,49);export let bgGreen=f(42,49);export let bgYellow=f(43,49);export let bgBlue=f(44,49);export let bgMagenta=f(45,49);export let bgCyan=f(46,49);export let bgWhite=f(47,49);export let bgGray=f(100,49)
```

#### -97B (-169B total):

```suggestion
import t from 'node:tty';let h=t.WriteStream.prototype.hasColors(),f=(s,e=39)=>h?v=>'\u001B['+s+'m'+v+'\u001B['+e+'m':v=>v,reset=f(0,0),bold=f(1,22),dim=f(2,22),italic=f(3,23),underline=f(4,24),overline=f(53,55),inverse=f(7,27),hidden=f(8,28),strikethrough=f(9,29),black=f(30),red=f(31),green=f(32),yellow=f(33),blue=f(34),magenta=f(35),cyan=f(36),white=f(37),gray=f(90),bgBlack=f(40,49),bgRed=f(41,49),bgGreen=f(42,49),bgYellow=f(43,49),bgBlue=f(44,49),bgMagenta=f(45,49),bgCyan=f(46,49),bgWhite=f(47,49),bgGray=f(100,49);export{reset,bold,dim,italic,underline,overline,inverse,hidden,strikethrough,black,red,green,yellow,blue,magenta,cyan,white,gray,bgBlack,bgRed,bgGreen,bgYellow,bgBlue,bgMagenta,bgCyan,bgWhite,bgGray}
```

#### -22B (-191B total)

```suggestion
import t from 'node:tty';let h=t.WriteStream.prototype.hasColors(),f=(s,e=39)=>h?v=>'\u001B['+s+'m'+v+'\u001B['+e+'m':v=>v,reset=f(0,0),bold=f(1,22),dim=f(2,22),i=f(3,23),u=f(4,24),o=f(53,55),j=f(7,27),h=f(8,28),s=f(9,29),black=f(30),red=f(31),green=f(32),y=f(33),blue=f(34),m=f(35),cyan=f(36),white=f(37),gray=f(90),b=f(40,49),bgRed=f(41,49),g=f(42,49),z=f(43,49),l=f(44,49),a=f(45,49),c=f(46,49),w=f(47,49),r=f(100,49);export{reset,bold,dim,i as italic,u as underline,o as overline,j as inverse,h as hidden,s as strikethrough,black,red,green,y as yellow,blue,m as magenta,cyan,white,gray,b as bgBlack,bgRed,g as bgGreen,z as bgYellow,l as bgBlue,a as bgMagenta,c as bgCyan,w as bgWhite,r as bgGray}
```

**Review by FlorianWendelborn — APPROVED** [review]

**Comment by FlorianWendelborn** [comment]
@fregante just updated the original comment, you could save another 9B by using the escaped Unicode character as a literal character instead

At that point, any hope of readability is lost though haha

**Comment by fregante** [comment]
Let's wait for Mr Sorrus to see if it's worth it

**Review by sindresorhus — APPROVED** [review]
💩

**Comment by fregante** [comment]
Approval was all I needed 😌

**Pull request closed without merging**